doc/development.md

```diff
@@ -35,24 +35,46 @@ Create virtual environment for vdsm:
 
 ## Building Vdsm
 
-To configure sources (run `./configure --help` to see configuration options):
+To configure sources (create Makefiles and configure script): 
 
     git clean -xfd
     ./autogen.sh --system --enable-timestamp
-    make
 
-To test Vdsm (refer to tests/README for further tests information):
+(It is recommended to start fresh after an update, from the previous step, before
+building RPMs, because the generated Makefile also contains the release number.)
 
-    make check
+Now you can create the virtual environment (https://docs.python.org/3/library/venv.html),
+which is necessary to run the tests later. This needs to be done only once:
+
+    make venv
+
+To see configuration options run `./configure -h`. Configure things, if necessary,
+then start make to build Vdsm:
+
+    make
 
-To create an RPM:
+To create the RPMs:
 
     make rpm
 
 To upgrade your system with local build's RPM:
 
     make upgrade
 
+To test Vdsm you need to activate the virtual environment first.
+(refer to [tests/README.md](tests/README.md) for further testing information):
+
+    . ~/.venv/vdsm/bin/activate
```

Comment:
Use source, "." works but cryptic.

Reply:
It may be POSIX and standard, but it is still cryptic and should not be used unless source is not available (this seems to be the case in FreeBSD).

Regardless, we should really use same instructions as in python venv documenation:
https://docs.python.org/3/library/venv.html#creating-virtual-environments